### PR TITLE
23 Add jdbc

### PR DESCRIPTION
**Pull request by Zejnilovic**
- adds the ability to use different JDBC drivers
- makes the loading of parameter more future/extension friendly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,12 +45,12 @@ Datasets Comparison
 Usage: spark-submit [spark options] --class za.co.absa.hermes.datasetComparison.DatasetComparisonJob hermes.jar [options]
 
   -f, --raw-format <value> format of the raw data (csv, xml, parquet,fixed-width, etc.)
-  --row-tag <value>        use the specific row tag instead of 'ROW' for XML format
+  --rowTag <value>        use the specific row tag instead of 'ROW' for XML format
```
Comment (yruslan) [inline README.md:48]: Why camelCase?

Comment (Zejnilovic) [reply README.md:48]: The official option from spark-xml is camel case. This is so I don't need to go through every possible option offered by different formats, parse is, etc. and this way I leverage the code that is already there.

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+case class DataframeOptions(format: String, options: Map[String, String], path: String) {
+  def loadDataFrame(implicit spark: SparkSession): DataFrame = {
+    val general = spark.read.format(format)
+    val generalWithOptions = if ( options.isEmpty ) general else general.options(options)
+
+    if (format == "jdbc") { generalWithOptions.load()     }
+    else                  { generalWithOptions.load(path) }
+  }
+}
```
Comment (felipemmelo) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala:20]: Maybe 2 more methods:

`private def setOptions(reader: DataReader, options: Map[String,String]): DataReader`

and 

`private def load(path: String, reader: DataReader): DataFrame`

then, the whole thing would be:

```
val general = spark.read.format(format)
val withOptions = setOptions(general, options)
load(path, withOptions)
```

**Review by felipemmelo — DISMISSED** [review]

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala (hunk 2 above)
Comment (Zejnilovic) [reply datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala:20]: Done

**Review by yruslan — APPROVED** [review]

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala [hunk 3]
```diff
@@ -32,44 +30,36 @@ object DatasetComparisonJob {
...
   def main(args: Array[String]): Unit = {
-    val cmd = DatasetComparisonConfig.getCmdLineArguments(args)  match {
-      case Success(value) => value
-      case Failure(exception) => throw exception
-    }
+    val cliOptions = CliOptions.parse(args)
 
     implicit val sparkSession: SparkSession = SparkSession.builder()
-      .appName(s"Dataset comparison - '${cmd.newPath}' and '${cmd.refPath}'")
+      .appName(s"Dataset comparison - '${cliOptions.referenceOptions.path}' and " +
```
Comment (wajda) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala:36]: bit: String concatenation

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala [hunk 4]
```diff
@@ -78,14 +68,18 @@ object DatasetComparisonJob {
...
+          expectedExceptActual.write.format("parquet").save(s"${cliOptions.outPath}/expected_minus_actual")
+          actualExceptExpected.write.format("parquet").save(s"${cliOptions.outPath}/actual_minus_expected")
       }
-      throw DatasetsDifferException(cmd.refPath, cmd.newPath, cmd.outPath, expectedDf.count(), actualDf.count())
+      throw DatasetsDifferException(cliOptions.referenceOptions.path,
+                                    cliOptions.newOptions.path,
+                                    cliOptions.outPath,
+                                    expectedDf.count(),
+                                    actualDf.count())
     }
```
Comment (wajda) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala:83]: nit: too much horizontal space. in multiline methods the every argument should be along on the line.

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import org.apache.spark.sql._
+
+case class DataframeOptions(format: String, options: Map[String, String], path: String) {
+
+  private def setOptions(dfReader: DataFrameReader): DataFrameReader =
+    if ( options.isEmpty ) dfReader else dfReader.options(options)
+
+  private def load(dfReader: DataFrameReader): DataFrame =
+    if (format == "jdbc") { dfReader.load()     }
+    else                  { dfReader.load(path) }
```
Comment (wajda) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala:13]: nit: Usage of the vertical alignment is not recommended

**Pull request merged**